### PR TITLE
wasm-smith: add configuration option to export everything

### LINKED ISSUE #654: `wasm-smith`: add method to export everything
Add a `export_everything` method to `Module` that does a post-generation pass to export everything in the module.

Various Wasmtime fuzz targets will call functions *if they are exported* or hash memories *if they are exported*.

An `export_everything` method would give us an easy way to export everything.

### PR DESCRIPTION
**Pull request by abrown**
Closes #654.

### CODE REVIEW

**Comment by abrown** [comment]
The issue mentions a post-processing pass but here it is implemented as a `Config` option; it will likely be easier to find this way.

File: crates/wasm-smith/src/core.rs [hunk 1]
```diff
@@ -918,36 +918,48 @@ impl Module {
...
+        if self.config.export_everything() {
+            for choices_by_kind in choices {
+                for (kind, idx) in choices_by_kind {
+                    let name = unique_string(1_000, &mut export_names, u)?;
+                    let ty = self.type_of(kind, idx);
+                    self.type_size += 1 + ty.size();
+                    self.exports.push((name, kind, idx));
                 }
+            }
+            Ok(())
```
Comment (fitzgen) [inline crates/wasm-smith/src/core.rs:930]: Can this be `return Ok(())` and then we can remove the `else` and the extra level of indentation below?

**Review by fitzgen — APPROVED** [review]
r=me with nitpick below addressed -- thanks!

**Comment by abrown** [comment]
Should be merge-able...

File: crates/wasm-smith/src/config.rs [hunk 2]
```diff
@@ -129,6 +129,12 @@ pub trait Config: 'static + std::fmt::Debug {
         100
     }
 
+    /// Export all WebAssembly objects in the module. This overrides
+    /// [`Config::min_exports`] and [`Config::max_exports`]. Defaults to false.
+    fn export_everything(&self) -> bool {
+        false
+    }
```
Comment (alexcrichton) [inline crates/wasm-smith/src/config.rs:136]: Could this be added to `SwarmConfig` as well? (probably `false` during the `Arbitrary` impl though)

**Review by alexcrichton — APPROVED** [review]

**Pull request merged**